File: TravellerCreator/ClientApp/src/app/sector-creator/models/star-frontiers/star-frontiers-planet.ts
import {IPlanet} from "../interfaces/planet";
import {Base} from "../enums/base";
import {TradeCode} from "../enums/trade-code";
import {TravelCode} from "../enums/travel-code";
import {PlanetType} from "../enums/planet-type";

export class StarFrontiersPlanet implements IPlanet{
  atmosphere: number;
  bases: Base[];
  government: number;
  hydrographics: number;
  lawLevel: number;
  name: string;
  population: number;
  size: number;
  starport: number;
  techLevel: number;
  temperature: number;
  tradeCodes: TradeCode[];
  travelCode: TravelCode;
  planetType: PlanetType;
  moons: StarFrontiersPlanet[];
}

File: TravellerCreatorModels/StarFrontiers/StarFrontiersPlanet.cs
using TravellerCreatorGlobalMethods;
using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.StarFrontiers;

public class StarFrontiersPlanet : IPlanet
{
    public string Name { get; set; } = "";
    public int Size { get; set; }
    public int Atmosphere { get; set; }
    public int Temperature { get; set; }
    public int Hydrographics { get; set; }
    public int Population { get; set; }
    public int Government { get; set; }
    public int LawLevel { get; set; }
    public char Starport { get; set; }
    public int TechLevel { get; set; }
    public List<TradeCode> TradeCodes { get; set; } = new();
    public List<Base> Bases { get; set; } = new();
    public TravelCode TravelCode { get; set; }
    public PlanetType PlanetType { get; set; }
    public List<IPlanet> Satellites { get; set; } = new();
}